src/core/store/modules/metadata/metadata.actions.ts

```diff
@@ -0,0 +1,58 @@
+import { createAction, createAsyncThunk } from '@reduxjs/toolkit';
+import { Interface } from '@ethersproject/abi';
+
+import { ThunkAPI } from '@frameworks/redux';
+
+const getABI = createAsyncThunk<{ ABI: string; Functions: string[] }, String, ThunkAPI>(
+  'metadata/getABI',
+  async (address, { extra, getState }) => {
+    const { onChainMetaDataService } = extra.services;
+    const { wallet } = getState();
+    const userAddress = wallet.selectedAddress;
+    const network = wallet.networkVersion;
+
+    if (!userAddress) throw new Error('WALLET NOT CONNECTED');
+
+    const OnChainMetaDataServiceResponse = await onChainMetaDataService.getContractABI(address, network!);
+    const ABI = OnChainMetaDataServiceResponse.data.result;
+
+    const iface = new Interface(OnChainMetaDataServiceResponse.data.result);
```

Comment:
what is `iface`? isnt ABI the only thing we need?

Reply:
was using it (abi -> interface) to get func names and store in state, but i guess i can do this logic in a util function and put func names in local state?